src/Domain/Collector.php

```diff
@@ -215,17 +219,17 @@ final class Collector
     /**
      * Creates a new instance of the Collector.
      *
-     * @param  string  $dir
+     * @param string $dir
+     * @param array<string> $files
      */
-    public function __construct(string $dir)
+    public function __construct(string $dir, array $files)
     {
         $this->dir = $dir;
+        $this->commonPath = $files !== [] ? PathShortener::extractCommonPath($files) : '';
```

Comment:
maybe we should pass the 'commonPath' instead of the array of files ?

Reply:
We pass and array of files to `PathShortener` and it will extract common path and get back to us. In case of no files (just for safety and not breaking the tests) common path will be empty string.